docs/authentication.md

```diff
@@ -0,0 +1,77 @@
+# Authentication
+
+This document details how to configure your Helios installation to authenticate
+requests between the Helios CLI and the Helios masters/cluster.
+
+# TODOs for this document
+
+- configuration needed to setup master to require authentication
+  - make sure authentication can be enabled based on client version header, to
```

Comment:
Another switch here when starting the master to make auth required for client versions greater than X? Defaults to requiring auth regardless of version if `--auth` is specified.